lib/matplotlib/_tight_layout.py

```diff
@@ -223,8 +223,13 @@ def get_subplotspec_list(axes_list, grid_spec=None):
     subplotspec_list = []
     for ax in axes_list:
         axes_or_locator = ax.get_axes_locator()
+
         if axes_or_locator is None:
             axes_or_locator = ax
+        elif hasattr(ax, '_twinned_axes'):
```

Comment:
The hasattr check is unnecessary, as it's a class variable.

Reply:
Oh I see.  Well, this fix here is therefore not very good logic.  Making the twinned axes siblings is problematic for the original issue - if we move one of the twins, the other twin is supposed to come along. But that implies one of the twins is the parent and one the child.  

I guess the redundant `_make_twin_axes` should remain for the two cases, and likely what needs to change is `set_postiion` to also change the position of any twins....

Reply:
> if we move one of the twins, the other twin is supposed to come along

But does it even make sense to call set_position on an axes that has an axes_locator set?  both are basically trying to force the axes' position.  Perhaps e.g. set_position could check if an axes_locator exists and ask the user to explicitly call `set_axes_locator(None)` first if they really want the axes to go here and nowhere else?  In that case, for twin axes, the correct way to set their position would be to set the "parent" axes' position, not the child one (which seems correct?) (Yes, I know, it's not a "parent-child" relationship in the usual sense.)

Reply:
Sure, OK, lets try this.  If an axes has a locator, but it is a twin, see if the twin has a locator; if not, and it has a subplotspec, use the subplotspec for tight layout.  This all seems a little more convoluted than it should be, but....